MyModels.py

```diff
@@ -0,0 +1,158 @@
+class BaseModel:
+    def __init__(self):
+        self.list = []
+        self.id_index = {}
+        
+    def add(self, entry):
+        
+        self.list.append(entry)
+    
+#     def find_index(self,entry):
+#         for i, item in self.list:
+#             if item==entry:
+#                 return i
+            
+#     def find_val(self,index):
+#         return self.list[index]
+
+#####################
+
+    def find_index(self,id):
+        return self.id_index[id]
+    
+    def find_val(self,id):
+        index = self.find_index(id)
+        return self.list[index]
+
+
+###################
+    def GetAll(self):
+        return self.list
+    
+    def delete(self, entry):
+        
+        self.list.remove(entry)
+    
+    def update(self,id, new_entry):
```

Comment:
Make sure you format your code. Here, there should be a space after the first comma: `self, id, ...`

Reply:
Sure, Thanks.